app/pages/settings/email.jsx

```diff
@@ -0,0 +1,289 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import talkClient from 'panoptes-client/lib/talk-client';
+import AutoSave from '../../components/auto-save';
+import handleInputChange from '../../lib/handle-input-change';
+
+class EmailSettingsPage extends React.Component {
+  constructor(props) {
+    super(props);
+    console.log(props)
+    this.state = {
+      page: 1,
+      projects: [],
+      projectPreferences: []
+    };
+    this.handlePreferenceChange = this.handlePreferenceChange.bind(this);
+    this.nameOfPreference = this.nameOfPreference.bind(this);
+    this.sortPreferences = this.sortPreferences.bind(this);
+    this.talkPreferenceOption = this.talkPreferenceOption.bind(this);
+    this.getProjectForPreferences(props.user);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (!this.state.page === prevState.page) {
+      this.getProjectForPreferences();
+    }
+  }
+
+  getProjectForPreferences(user) {
+    return user.get('project_preferences', {
+      page: this.state.page,
+      sort: 'display_name'
+    })
+    .then((preferences) => {
+      if (preferences.length > 0) {
+        const projectIDs = preferences.map(pref => pref.links.project);
+        apiClient.type('projects').get({
+          id: projectIDs
+        })
+        .then((projects) => {
+          return this.setState({
+            meta: preferences[0].getMeta(),
+            projectPreferences: preferences,
+            projects
+          });
+        });
+      }
+    })
+    .catch((error) => {
+      console.log('Something went wrong. Error: ', error);
+    });
+  }
+
+  handleProjectEmailChange(projectPreference, args) {
+    handleInputChange.apply(projectPreference, args);
+    projectPreference.save();
+    this.setState();
+  }
+
+  handlePreferenceChange(preference, event) {
+    return preference.update({
+      email_digest: event.target.value
+    })
+    .save();
+  }
+
+  nameOfPreference(preference) {
+    switch (preference.category) {
+      case 'participating_discussions':
+        'When discussions I\'m participating in are updated';
+        break;
+      case 'followed_discussions':
+        'When discussions I\'m following are updated';
+        break;
+      case 'mentions':
+        'When I\'m mentioned';
+        break;
+      case 'group_mentions':
+        'When I\'m mentioned by group (this.admins, this.team, etc)';
+        break;
+      case 'messages':
+        'When I receive a private message';
+        break;
+      case 'started_discussions':
+        'When a discussion is started in a board I\'m following';
+        break;
+      default:
+        'nope';
+
+    }
+  }
+
+  renderProjectPreferences() {
+    const { projects, projectPreferences } = this.state;
+    return projectPreferences.map((projectPreference, i) => {
+      if (projects[i]) {
+        return (
+          <tr key={i}>
```

Comment:
[Don't use array indexes as React keys.](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318) `project.id` or `projectPreference.id` would be better here.